src/sys/unix/kqueue.rs

```diff
@@ -343,7 +343,12 @@ pub mod event {
             (event.flags & libc::EV_EOF) != 0 && event.fflags != 0
     }
 
-    pub fn is_hup(event: &Event) -> bool {
+    pub fn is_hup(_: &Event) -> bool {
+        // Not supported.
+        false
+    }
+
+    pub fn is_read_hup(event: &Event) -> bool {
         (event.flags & libc::EV_EOF) != 0
```

Comment:
This needs to also check, `event.filter == libc::EVFILT_READ`. Sorry about that, I should have seen it.